runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingSpotOverlay.java

```diff
@@ -57,6 +61,9 @@
 	private final Client client;
 	private final ItemManager itemManager;
 
+	private static HashMap<WorldPoint, HashMap<FishingSpot, NPC>> SPOTS_LIST = new HashMap<>();
```

Comment:
Just make new type and use 1 map

Reply:
I've gone ahead and created a new inner class to take care of that. Each instance of the class FishingSpotInfo will store 3 ArrayLists, containing a list of FishingSpots, NPCs, and counts. The getters were implemented using indexes to maintain the same lookup speed as before.

I've tested these new changes and they're functioning as expected.